api/src/main/java/io/druid/data/input/FirehoseFactory.java

```diff
@@ -36,7 +37,13 @@
    * If this method returns null, then any attempt to call hasMore(), nextRow(), commit() and close() on the return
    * value will throw a surprising NPE.   Throwing IOException on connection failure or runtime exception on
    * invalid configuration is preferred over returning null.
+   * <p/>
+   * Some fire hoses like {@link PrefetchableTextFilesFirehoseFactory} may use a temporary
+   * directory to cache data in it.
+   *
+   * @param parser             an input row parser
+   * @param temporaryDirectory a directory where temporary files are stored
    */
-  public Firehose connect(T parser) throws IOException, ParseException;
+  Firehose connect(T parser, File temporaryDirectory) throws IOException, ParseException;
```

Comment:
The life cycle of temporaryDirectory is same with that of FirehoseFactory rather than Firehose. This is because the temporaryDirectory containing cached data should not be deleted for reusing even when a PrefetchableFirehose is closed. 

We may add abstract methods for initialization and termination to FirehoseFactory and the implementation of PrefetchableFirehoseFactory can create and delete the temporaryDirectory in those methods. Do you think it is better?